mlir/python/mlir/dialects/func.py

```diff
@@ -105,6 +105,10 @@ def add_entry_block(self, arg_locs: Optional[Sequence[Location]] = None):
 
     @property
     def arg_attrs(self):
+        if ARGUMENT_ATTRIBUTE_NAME not in self.attributes:
+            self.attributes[ARGUMENT_ATTRIBUTE_NAME] = ArrayAttr.get(
```

Comment:
I think it makes more sense to just return `None` here? I know it doesn't match the correct semantic but it seems odd to make empty dictionaries just to return them as empty?

Reply:
I can see your point, however in the code I was working on, I have something like:

```python
for arg, arg_attr in zip(block.arguments, func_op.arg_attrs):
    if "my.value" in arg_attr:
        lst1.append(arg)
    else:
        lst2.append(arg)
```

In your proposal, I would need to add a `None` check such as:

```python
if func_op.arg_attrs is None:
    lst2.extend(block.arguments)
else:
    for arg, arg_attr in zip(block.arguments, func_op.arg_attrs):
        if "my.value" in arg_attr:
            lst1.append(arg)
        else:
            lst2.append(arg)
```

This would improve performance, but this is Python and perhaps not the chief concern, rather simplicity.

Happy to make the change to return `None` if you prefer, but from my perspective the overhead of generating empty dicts and then walking over an empty list makes more sense.